src/relayserver/RelayServer.ts

```diff
@@ -209,26 +226,28 @@ export class RelayServer extends EventEmitter {
         }
         throw new Error(message)
       }
-      acceptanceBudget = this.config.maxAcceptanceBudget
-      const paymasterAcceptanceBudget = parseInt(gasLimits.acceptanceBudget)
-      if (paymasterAcceptanceBudget > acceptanceBudget) {
+      const paymasterAcceptanceBudget = parseInt(gasAndDataLimits.acceptanceBudget)
+      // TODO remove, since it's redundant. This check is also done in relayCall() on-chain, so the server will fail
+      // on view call if these requirements aren't met.
+      if (paymasterAcceptanceBudget + dataGasCost > relayExposure) {
         if (!this._isTrustedPaymaster(paymaster)) {
           throw new Error(
-            `paymaster acceptance budget too high. given: ${paymasterAcceptanceBudget} max allowed: ${this.config.maxAcceptanceBudget}`)
+            `paymaster acceptance budget too high. given: ${paymasterAcceptanceBudget} max allowed: ${this.config.maxRelayExposure}`)
```

Comment:
also depends on `dataGasCost`, isn't it?

Reply:
Right.